pandas/io/formats/style_render.py

```diff
@@ -495,9 +493,12 @@ def format(
                 escape=escape,
             )
 
-            for row, value in data[[col]].itertuples():
-                i, j = self.index.get_loc(row), self.columns.get_loc(col)
-                self._display_funcs[(i, j)] = format_func
+            for row in data[[col]].itertuples():
+                i_ = self.index.get_indexer_for([row[0]])  # handle duplicate keys in
+                j_ = self.columns.get_indexer_for([col])  # non-unique indexes
```

Comment:
you can do this outside of the loop right? (as col doesn't change).

does this change perf at all? (I don't think so, but checking).

Reply:
Good catch.. the multiple loops really killed it for the unique case (which is benchmarked).. 

```
       before           after         ratio
     [4af3eed5]       [3a8f11e5]
     <styler_non_unique~1^2>       <styler_non_unique>
+        57.8±2ms          163±4ms     2.82  io.style.Render.time_format_render(24, 120)
+        87.1±5ms          242±4ms     2.78  io.style.Render.time_format_render(36, 120)
+      30.5±0.9ms       82.2±0.4ms     2.69  io.style.Render.time_format_render(12, 120)
+     8.53±0.08ms       14.7±0.2ms     1.72  io.style.Render.time_format_render(12, 12)
+      16.2±0.3ms       27.7±0.7ms     1.71  io.style.Render.time_format_render(24, 12)
+        25.3±1ms       40.9±0.7ms     1.62  io.style.Render.time_format_render(36, 12)
+      16.6±0.2ms       18.5±0.2ms     1.11  io.style.Render.time_classes_render(36, 12)

SOME BENCHMARKS HAVE CHANGED SIGNIFICANTLY.
PERFORMANCE DECREASED.
```

So I had to separate out the non-unique and unique cases with a conditional, then performance was the same...

```
BENCHMARKS NOT SIGNIFICANTLY CHANGED.
```